### PR TITLE
[Perf]Enable regional compile for Cosmos3 HSDP blocks

### PR DESCRIPTION
**Pull request by bjf-frz**
PLEASE FILL IN THE PR DESCRIPTION HERE.

## Purpose

#4340 

  Enable regional torch.compile to match Cosmos3 denoising blocks under HSDP.

  Cosmos3 declares Cosmos3GenDecoderLayer as its repeated block, but after HSDP wrapping, the module class name no longer matches the original block class name. As a result, regional compilation was skipped even though the model runner attempted to compile the transformer.

  This change keeps the existing class-name matching path, adds common wrapper unwrapping checks, and falls back to Cosmos3's stable gen_layers.<idx> module path for HSDP-wrapped denoising blocks.

## Test Plan
Cosmos3 I2V 35-step smoke benchmark, 4 GPUs, cfg2 + usp2 + vaepp4 + hsdp

  Setup:

  - Model: nvidia/Cosmos3-Nano
  - Task: I2V
  - Resolution: 1280x720
  - Frames: 189
  - Steps: 35
  - Guidance scale: 6.0
  - Seed: 1234
  - Parallel config: cfg_parallel_size=2, usp=2, vae_patch_parallel_size=4, hsdp_shard_size=4

| Build | wall time |
| -- | -- |
| Before compiled | 275.438s |
| After compiled | 268.396s |

The change improves wall time by ~2.6% in this smoke run.

**vLLM Version:**

**vLLM-Omni Commit:**

## Test Result

before:

https://github.com/user-attachments/assets/c4ab5b76-f709-46d2-8263-e38c90c173f0

fixed:

https://github.com/user-attachments/assets/8e79fd0d-8394-4caa-8fe6-276ab39a907d

To satisfy the requirement that each size of video should be under 10Mb, the output was idited in 5s.

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
check the profiling please

**Review by hsliuustc0106 — COMMENTED** [review]
Perf optimization with benchmarks included.

**Review by RuixiangMa — COMMENTED** [review]
This bug can affects all models with both _repeated_blocks and _hsdp_shard_conditions. Can you verify whether the other  models have the same issue?

File: vllm_omni/diffusion/compile.py [hunk 1]
```diff
@@ -8,6 +8,25 @@
 logger = init_logger(__name__)
 
 
+def _matches_repeated_block(name: str, module: nn.Module, repeated_blocks: list[str]) -> bool:
```
Comment (david6666666) [inline vllm_omni/diffusion/compile.py]: Please add a small unit test for this helper with a wrapped `gen_layers.0` module and a non-target wrapper. The current coverage does not exercise `regionally_compile`, so this matching path can regress without CI catching it.

File: vllm_omni/diffusion/compile.py [hunk 2]
```diff
@@ -8,6 +8,25 @@
...
+
+    for attr in ("_fsdp_wrapped_module", "module", "_orig_mod"):
+        wrapped = getattr(module, attr, None)
+        if wrapped is not None and wrapped.__class__.__name__ in repeated_blocks:
+            return True
+
+    parts = name.split(".")
+    return (
+        "Cosmos3GenDecoderLayer" in repeated_blocks
+        and len(parts) >= 2
```
Comment (david6666666) [inline vllm_omni/diffusion/compile.py]: Can we avoid hard-coding Cosmos3 here? `regionally_compile()` is shared by every model with `_repeated_blocks`; this should come from model metadata or the existing HSDP/block-container matching path so the next wrapped model does not need another special case.

**Comment by bjf-frz** [comment]
> This bug can affects all models with both _repeated_blocks and _hsdp_shard_conditions. Can you verify whether the other models have the same issue?

  Thanks for the callout. I generalized the fallback matching logic so it is no longer Cosmos3-specific.

  Instead of hardcoding `Cosmos3GenDecoderLayer` / `gen_layers`, regional compile now uses the model-declared `_layerwise_offload_blocks_attrs` to match wrapped repeated blocks by their container path, e.g.:

  - Cosmos3: `gen_layers.<idx>`
  - Qwen-Image: `transformer_blocks.<idx>`
  - Wan/LTX-style models: `blocks.<idx>` or `transformer_blocks.<idx>` when declared by the model

  This keeps the behavior tied to each model’s own repeated-block metadata instead of adding model-specific checks in `compile.py`.

File: vllm_omni/diffusion/compile.py (hunk 1 above)
Comment (bjf-frz) [reply vllm_omni/diffusion/compile.py]: > Please add a small unit test for this helper with a wrapped `gen_layers.0` module and a non-target wrapper. The current coverage does not exercise `regionally_compile`, so this matching path can regress without CI catching it.

I added a unit test covering the HSDP-wrapped case where the module class name no longer matches `_repeated_blocks`, but the module is still under a declared
  repeated-block container.

File: vllm_omni/diffusion/compile.py (hunk 2 above)
Comment (bjf-frz) [reply vllm_omni/diffusion/compile.py]: > Can we avoid hard-coding Cosmos3 here? `regionally_compile()` is shared by every model with `_repeated_blocks`; this should come from model metadata or the existing HSDP/block-container matching path so the next wrapped model does not need another special case.

Instead of hardcoding `Cosmos3GenDecoderLayer` / `gen_layers`, regional compile now uses the model-declared `_layerwise_offload_blocks_attrs` to match wrapped repeated blocks by their container path.

**Comment by david6666666** [comment]
LGTM

**Review by david6666666 — APPROVED** [review]
